### PR TITLE
Fix Elasticsearch instrumentation failing due to event payloads being multiple types

### PR DESCRIPTION
**Pull request by ejsmith**

### CODE REVIEW

File: src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/MultiTypePropertyFetcher.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+namespace OpenTelemetry.Instrumentation
+{
+    /// <summary>
+    /// PropertyFetcher fetches a property from an object.
+    /// </summary>
+    /// <typeparam name="T">The type of the property being fetched.</typeparam>
+    public class MultiTypePropertyFetcher<T>
+    {
+        private readonly string propertyName;
+        private ConcurrentDictionary<Type, PropertyFetch> innerFetcher = new ConcurrentDictionary<Type, PropertyFetch>();
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/MultiTypePropertyFetcher.cs:31]: nit: You can add readonly here.

File: src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/MultiTypePropertyFetcher.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        /// <returns>Property fetched.</returns>
+        public T Fetch(object obj)
+        {
+            if (obj == null)
+            {
+                return default(T);
+            }
+
+            var type = obj.GetType().GetTypeInfo();
+            var fetcher = this.innerFetcher.GetOrAdd(type, t =>
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/MultiTypePropertyFetcher.cs:55]: One small perf improvement you could make, don't use GetOrAdd. The lambda will always allocate a delegate (even if it isn't called). If you use TryGetValue and then TryAdd (if nothing was found) you can avoid that.

**Review by CodeBlanch — APPROVED** [review]
Sent a couple of suggestions but LGTM

File: src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/MultiTypePropertyFetcher.cs (hunk 1 above)
Comment (ejsmith) [reply src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/MultiTypePropertyFetcher.cs:31]: Done

File: src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/MultiTypePropertyFetcher.cs (hunk 2 above)
Comment (ejsmith) [reply src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/MultiTypePropertyFetcher.cs]: Done

**Comment by ejsmith** [comment]
@CodeBlanch anything else I need to do for this?

**Pull request merged**